docs/development/extensions-contrib/k8s-jobs.md

```diff
@@ -115,33 +75,197 @@ You can keep your Dockerfile the same but you must have a sidecar spec like so:
+For both of these adapters, you can add optional labels to your K8s jobs / pods if you need them by using the following configuration:
+`druid.indexer.runner.labels: '{"key":"value"}'`
+Annotations are the same with:
+`druid.indexer.runner.annotations: '{"key":"value"}'`
+
+All other configurations you had for the middle manager tasks must be moved under the overlord with one caveat, you must specify javaOpts as an array:
+`druid.indexer.runner.javaOptsArray`, `druid.indexer.runner.javaOpts` is no longer supported.
+
+If you are running without a middle manager you need to also use `druid.processing.intermediaryData.storage.type=deepstore`
+
+### Custom Template Pod Adapter
+The custom template pod adapter allows you to specify a pod template file per task type for more flexibility on how to define your pods. This adapter expects a [Pod Template](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates) to be available on the overlord's file system. This pod template is used as the base of the pod spec for the Kubernetes Job. You can override things like labels, environment variables, resources, annotation, or even the base image with this template. To enable this pod adapter you can specify the runtime property `druid.indexer.runner.k8s.adapter.type: customTemplateAdapter`
+
+The base pod template must be specified as the runtime property `druid.indexer.runner.k8s.podTemplate.base: /path/to/basePodSpec.yaml`
+
+Task specific pod templates must be specified as the runtime property `druid.indexer.runner.k8s.podTemplate.{taskType}: /path/to/taskSpecificPodSpec.yaml` where {taskType} is the name of the task type i.e `index_parallel`
+
+The following is an example Pod Template that uses the regular druid docker image.
+```
+apiVersion: "v1"
+kind: "PodTemplate"
+template:
+  metadata:
+    annotations:
+      custom-annotation: "hello"
+    labels:
+      custom-label: "hello"
+  spec:
+    affinity: {}
+    containers:
+    - command:
+        - sh
+        - -c
+        - |
+          /peon.sh /druid/data 1
+      env:
+      - name: CUSTOM_ENV_VARIABLE
+        value: "hello"
+      image: apache/druid:26.0.0
+      name: main
+      ports:
+      - containerPort: 8091
+        name: druid-tls-port
+        protocol: TCP
+      - containerPort: 8100
+        name: druid-port
+        protocol: TCP
+      resources:
+        limits:
+          cpu: "1"
+          memory: 2400M
+        requests:
+          cpu: "1"
+          memory: 2400M
+      volumeMounts:
+      - mountPath: /opt/druid/conf/druid/cluster/master/coordinator-overlord # runtime props are still mounted in this location because that's where peon.sh looks for configs
+        name: nodetype-config-volume
+        readOnly: true
+      - mountPath: /druid/data
+        name: data-volume
+      - mountPath: /druid/deepstorage
+        name: deepstorage-volume
+    restartPolicy: "Never"
+    securityContext:
+      fsGroup: 1000
+      runAsGroup: 1000
+      runAsUser: 1000
```

Comment:
Can you explain what these settings are please

Reply:
this is the user the druid process runs on. i think we specify 1000 here because that user is created in the druid docker image that i used in this example.